src/Aspire.Cli/Builds/AppHostBuilder.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Cli.Builds;
+
+internal interface IAppHostBuilder
+{
+    Task<int> BuildAppHostAsync(FileInfo projectFile, bool useCache, CancellationToken cancellationToken);
+}
+
+internal sealed class AppHostBuilder(ILogger<AppHostBuilder> logger, IDotNetCliRunner runner) : IAppHostBuilder
+{
+    private readonly ActivitySource _activitySource = new ActivitySource(nameof(AppHostBuilder));
+    private readonly SHA256 _sha256 = SHA256.Create();
+
+    private async Task<string> GetBuildFingerprintAsync(FileInfo projectFile, CancellationToken cancellationToken)
+    {
+        using var activity = _activitySource.StartActivity();
+
+        _ = logger;
+
+        var msBuildResult = await runner.GetProjectItemsAndPropertiesAsync(
```

Comment:
How do we feel about transitivity here? You might need the same finger print for the closure of project refs (that aren't aspire projects).